### PR TITLE
add dat-json into dat-node

### PR DESCRIPTION
**Pull request by okdistribute**
now `dat.meta.read` and `dat.meta.write` are available to deal with the dat.json file in the repository.

### CODE REVIEW

**Comment by karissa** [comment]
weird, why does the pr fail but the push doesnt? @joehand

**Comment by joehand** [comment]
> weird, why does the pr fail but the push doesnt? @joehand

Hmm, its some pesky peer count tests. I thought I put enough leeway into them but I guess I need some more. I ran it again and its passing but I'll look at those tests again. 

Thanks for doing this! Can we add some tests and docs for this too?

**Comment by juliangruber** [comment]
I guess this would also need a `dat.meta.write()` on archive creation?

File: lib/dat-json.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+var stringKey = require('dat-encoding').toStr
+var path = require('path')
+
+module.exports = function (dat) {
+  return {
+    read: function (cb) {
+      // dat.json
+      // reads to dat.meta if exists
+      // (TODO: move to module & validate dat.json)
+      fs.readFile(datJsonFile(dat), 'utf8', function (err, body) {
```
Comment (juliangruber) [inline lib/dat-json.js]: Should this go to disk, or rather read meta from an archive live stream? This way we could also implement a follow mode where you subscribe to meta updates.

**Pull request closed without merging**